main/js/afterSubmit.js

```diff
@@ -187,16 +187,28 @@ const buildHeatmap = function(expData, clinData) {
  */
 const buildViolinPlot = function(geneQuery, expressionData) {
   //Remove loader from violin plot container
-  document.getElementById("violinLoaderDiv").classList.remove("loader");
+  var violinLoaderDiv = document.getElementById("violinLoaderDiv");
+  violinLoaderDiv.classList.remove("loader");
 
-  // Create partition selector
-  const partitionDivId = "violinPartition";
-  addDivInside(partitionDivId, "violinLoaderDiv");
-  // Create partition selector
-  createViolinPartitionBox("violinPlots", geneQuery);
+  //Setup Materialize Grid
+  addDivInside("violinGridRow", violinLoaderDiv.id);
+  var gridRow = document.getElementById("violinGridRow");
+  gridRow.classList.add("row");
 
-  // Create div for violin plots
-  addDivInside("violinPlots", "violinLoaderDiv");
+  // Create partition selector div and add it inside Materialize Grid
+  const partitionDivId = "violinPartition";
+  addDivInside(partitionDivId, gridRow.id);
+  var partitionCol = document.getElementById(partitionDivId);
+  partitionCol.classList.add("col");
+  partitionCol.classList.add("s3");
+  //Generate the partition selector
+  createViolinPartitionBox(partitionDivId, geneQuery);
+
+  // Create div for violin plots and add it inside Materialize Grid
+  addDivInside("violinPlots", gridRow.id);
+  var violinPlotsCol = document.getElementById("violinPlots");
+  violinPlotsCol.classList.add("col")
+  violinPlotsCol.classList.add("s8");
```

Comment:
similar comment to above

```suggestion
  violinPlotsCol.classList.add("col", "s8");
```